api/src/test/java/org/openmrs/util/OpenmrsUtilTest.java

```diff
@@ -217,6 +219,15 @@ public void validatePassword_shouldFailWithoutUpperAndLowerCasePasswordIfNotAllo
 		OpenmrsUtil.validatePassword("admin", "test0nl1", "1-8");
 	}
 	
+	/**
+	 * @see OpenmrsUtil#conceptListHelper(String)
+	 */
+	@Test
+	public void conceptListHelper_shouldReturnListOfUniqueConceptsWithGivenNameInDescriptor() {
+		List<Concept> helperList = OpenmrsUtil.conceptListHelper("name:YES|name:YES");
+		assertTrue("The list should have just one element" , helperList.size() == 1);
```

Comment:
Better would be a test that also asserts what the expected list is made of, rather than just checking its size. There are many ways to have a size 1 array that is not the one we hope for.

I see that tests are simply missing basically for those two methods, they just didn't exist in the first place. So regular tests will have to be added too, not just the edge cases mentioned in the todo's.

Also those tests should be in `OpenmrsUtilUnitTest` with `ConceptService` being mocked. It is overkill to load the whole Spring Test context just to test the behaviour of such util methods (as you saw explained [here](https://talk.openmrs.org/t/a-quick-question-about-testing-some-modules/21547/9?u=mksd)).

Reply:
I wrote tests for them in this PR https://github.com/openmrs/openmrs-core/pull/2820. Should we finish this first ? Otherwise the other PR will be less relevant .

Reply:
@Andu033 yes, tests **and** JavaDoc.

Reply:
How to mock the ConceptService?